### ISSUE TITLE
`JLOptions.julia_bin` incorrect when embedding

### ISSUE THREAD

**Issue by twadleigh** [JuliaLang/julia#18437]
For instance, inserting

``` C
jl_eval_string("println(Base.JLOptions())");
```

into the embedding example yields in part:

```
  julia_bin = "C:\\tw\\julia\\usr\\bin\\embedding.exe",
```

the embedding executable, and not the julia exectuable.

**Comment by yuyichao** [comment]
IMHO, this seems like the right thing to do.

**Comment by twadleigh** [comment]
@yuyichao, except that at one point in the past I seem to recall @vtjnash saying that that field was supposed to be the path to the julia executable for the purpose of spawning other julia processes ala distributed computing. That may or may not be the case, however, if it is, in fact, meant to represent the path of the executable, `julia` or otherwise, then, at the very least, it is mis-named.

My real motive in opening this issue, however, is less about the interpretation/calculation of this field and more to lay more solid groundwork for justifying some tweaks to the way that julia does path-guessing in `jl_init` that generalizes to the embedding case (cf. #18421).

**Comment by yuyichao** [comment]
> the path to the julia executable for the purpose of spawning other julia processes ala distributed computing

No. We are not using it for that purpose (AFAIK no one is using it).

**Comment by twadleigh** [comment]
@yuyichao I didn't think we were. I do know for a fact that there is not a single reference to that field (outside where it is defined and set) anywhere in the julia repo. Of course, in that case, it begs the question as to whether that field should even exist.

What I'm really doing, though, is preempting at least one reason that was given at one point for not preferring tweaks to path-guessing that work for both the usual and the embedding cases.

**Comment by yuyichao** [comment]
What `julia_bin` contains doesn't have to be related to how we find `JULIA_HOME`. It does seem that the value is never used anywhere directly and can probably be removed.

I know there's multiple ways to query the current executable name, libuv also provides the wrapper for that, which is probably why it is used. Is there an easy way to figure out the full path of the current library?

**Comment by twadleigh** [comment]
In short, yes.

**Comment by ihnorton** [comment]
Fixed by #21299, #21427, and family. If this is still an issue, comment and we can reopen.
